api/managementpb/ia/rules.proto

```diff
@@ -0,0 +1,202 @@
+import "google/api/annotations.proto";
+import "google/protobuf/duration.proto";
+import "google/protobuf/timestamp.proto";
+import "managementpb/ia/boolean_flag.proto";
+import "managementpb/ia/channels.proto";
+import "managementpb/ia/params.proto";
+import "managementpb/ia/templates.proto";
+import "managementpb/severity.proto";
+
+// TODO move it to parent directory / package once API is v1-stable
+
+// FilterType represents filter matching type.
+enum FilterType {
+  FILTER_TYPE_INVALID = 0;
+  // =
+  EQUAL = 1;
+  // !=
+  NOT_EQUAL = 2;
+  // =~
+  REGEX = 3;
+  // !~
+  NOT_REGEX = 4;
+}
+
+// Filter repsents a single filter condition.
+message Filter {
+  FilterType type = 1;
+  string key = 2;
+  string value = 3;
+}
+
+// RuleParam represents a single rule parameter for List, Change and Update APIs.
+message RuleParam {
+  // Machine-readable name (ID) that is used in expression.
+  string name = 1 [
+    (validator.field) = {
+      string_not_empty: true
+    }
+  ];
+  // Parameter type.
+  ParamType type = 2;
+  // Parameter value.
+  oneof value {
+    // Bool value.
+    bool bool = 3;
+    // Float value.
+    float float = 4;
+    // String value.
+    string string = 5;
+  }
+}
+
+// Rule represents Alert Rule.
+message Rule {
+  // Template used to create that rule.
+  Template template = 1;
+  // Rule ID.
+  string rule_id = 2;
+  // True if that rule is disabled.
+  bool disabled = 3;
+  // Rule human-readable summary.
+  string summary = 4;
+  // Rule parameters.
+  repeated RuleParam params = 5;
+  // Rule duration.
+  google.protobuf.Duration for = 6;
+  // Rule severity.
+  management.Severity severity = 7;
+  // Custom labels.
+  map<string, string> custom_labels = 8;
+  // Filters.
+  repeated Filter filters = 9;
+  // Channels.
+  repeated Channel channels = 10;
+  // Rule creation time.
+  google.protobuf.Timestamp created_at = 11;
+}
+
+message ListAlertRulesRequest {
+  // If true, rule files will be re-read from disk.
+  bool reload = 1;
+}
+
+message ListAlertRulesResponse {
+  repeated Rule rules = 1;
+}
+
+message CreateAlertRuleRequest {
+  // Template name.
+  string template_name = 1;
+  // New rule status.
+  bool disabled = 2;
+  // Rule parameters. All template parameters should be set.
+  repeated RuleParam params = 3;
+  // Rule duration. Should be set.
+  google.protobuf.Duration for = 4;
+  // Rule severity. Should be set.
+  management.Severity severity = 5;
+  // All custom labels to add or remove (with empty values) to default labels from template.
+  map<string, string> custom_labels = 6;
+  // Filters. Should be set.
+  repeated Filter filters = 7;
+  // Channels. Should be set.
+  repeated string channel_ids = 8;
+}
+
+message CreateAlertRuleResponse {}
+
+message UpdateAlertRuleRequest {
+  // Rule ID.
+  string rule_id = 1 [
+    (validator.field) = {
+      string_not_empty: true
+    }
+  ];
+  // New rule status. Should be set.
```

Comment:
Not new

Reply:
Status is new :)

Reply:
But what if I don't change status, then it's old) Nevermind 😄